pkg/controller/clusterconnection/clusterconnection_controller.go

```diff
@@ -331,16 +331,29 @@ func (r *ReconcileConnection) Reconcile(ctx context.Context, request reconcile.R
 		}
 	}
 
+	// Populate a list of namespaces to be displayed in the service graph Tigera infrastructure layer.
+	sgLayerTigeraNameSpaces := render.DefaultSGLayerTigeraNamespaces()
+	if !sgLayerTigeraNameSpaces[render.GuardianNamespace] {
+		sgLayerTigeraNameSpaces[render.GuardianNamespace] = true
+	}
+	amz, err := utils.GetAmazonCloudIntegration(ctx, r.Client)
+	if err != nil {
```

Comment:
If there is no AmazonCloudIntegration in the cluster, then err will not be nil. I think you want to filter out the NotFound error. If NotFound then we do nothing. If it is another error, we need to SetDegraded()

Reply:
Done